sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -498,13 +498,13 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       origin = t.origin)
   }
 
-  def expectViewNotTableError(
+  def unsupportedTableOperationError(
```

Comment:
There is already a `unsupportedTableOperationError` in this file, shall we merge these two?

Reply:
Let me take a look.

Reply:
Well, this is the legacy issue, 
The error class used by the another ·unsupportedTableOperationError` method is `UNSUPPORTED_FEATURE.TABLE_OPERATION`. 

The biggest difference between it and the current `UNSUPPORTED_TABLE_OPERATION.WITH_SUGGESTION` is that the prompt is different. 

- The `UNSUPPORTED_FEATURE.TABLE_OPERATION` is
https://github.com/apache/spark/blob/b6190a3db974c19b6a0c4fe7af75531d67755074/common/utils/src/main/resources/error/error-classes.json#L3218-L3222

- The `UNSUPPORTED_TABLE_OPERATION.WITH_SUGGESTION` is:
https://github.com/apache/spark/blob/b6190a3db974c19b6a0c4fe7af75531d67755074/common/utils/src/main/resources/error/error-classes.json#L3440-L3444

### Let me try to merge them together in this PR.

Reply:
I think a good reason to move a sub error class to top-level error class is itself needs sub error classes.